components/suggestion/search_suggestion_tagged_input.jsx

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import * as Utils from 'utils/utils.jsx';
+
+import SearchSuggestionInput from './search_suggestion_input';
+const exampleTags = ['from:', 'in:', 'on:', 'before:', 'after:'];
+
+const includesTags = (str) => {
+    const lcStr = str.toLowerCase();
+    return exampleTags.some((tag) => lcStr.includes(tag));
+};
+
+const isTag = (str) => {
+    const lcStr = str.toLowerCase();
+    return exampleTags.some((tag) => lcStr.startsWith(tag));
+};
+
+const determinePairs = (value = this.props.value) => {
+    const pairIncludesTags = includesTags(value);
+    let remainder;
+    let pairs = [];
+
+    if (pairIncludesTags) {
+        const split = value.split(/\s(?=\w+:)/);
+        let lastEntry = split[split.length - 1];
```

Comment:
We might've missed this in the previous version, but what if earlier entries aren't tags and values like you search for `apples in:town-square`?

Reply:
then the remainder will be the value and thus appear as the value of the input.

Reply:
Looks like this actually works fine if text appears before a tag.

![Screen Shot 2020-06-19 at 12 19 06 PM](https://user-images.githubusercontent.com/3277310/85155718-18eee700-b227-11ea-8253-6e5b2fe79777.png)